FAQ.md

```diff
@@ -0,0 +1,135 @@
+# Frequently Asked Questions
+
+## Design
+
+### Why spend so much effort on logger performance?
+
+Of couse, most applications won't notice the impact of a slow logger: they
+already take tens or hundreds of milliseconds for each operation, so an extra
+millisecond doesn't matter.
+
+On the other hand, why *not* make structured logging fast? The `SugaredLogger`
+isn't any harder to use than other logging packages, and the `Logger` makes
+structured logging possible in performance-sensitive contexts. Across a fleet
+of Go microservices, making each application even slightly more efficient adds
+up quickly.
+
+### Why aren't `Logger` and `SugaredLogger` interfaces?
+
+Unlike most widely used interfaces, both `Logger` and `SugaredLogger`
+interfaces would include *many* methods. As [Rob Pike points
+out][go-proverbs], "The bigger the interface, the weaker the abstraction."
+Interfaces are also rigid &mdash; *any* change requires releasing a new major
+version, since it breaks all third-party implementations.
+
+Making the `Logger` and `SugaredLogger` concrete types doesn't sacrifice much
+abstraction, and it lets us add methods without introducing breaking changes.
+Your applications can and should define and depend upon an interface that
+includes just the methods you use.
+
+### Why sample application logs?
+
+Applications often experience runs of errors, either because of a bug or
+because of a misbehaving user. Logging errors is usually a good idea, but it
+can sometimes make a bad situation worse: not only is your application coping
+with an increased number of errors, it's also spending extra CPU cycles and
+I/O logging those errors. In the worst case, each call to your logger is
+contending over the same mutex, limiting throughput when you need it most.
```

Comment:
Rather than calling out a mutex, I think it's better to say something like "since writes are typically serialized, [...]". Since the cost of the mutex is actually pretty negligible, it's that we are blocked till the writes are done which ends up serializing all log calls